### DISCUSSION TITLE
Adding a git remote or cloning with SSH on a non-standard port

### DISCUSSION THREAD

**Discussion by kastl-ars** [jj-vcs/jj#5147] [Q&A] [answered]
Dear all,

I am starting to use jujutsu and find it very nice indeed. I could clone from Codeberg/Github/etc. successfully

However, I cannot clone a repository from my private Forgejo instance using SSH on a non-standard port:

```
$ jj git clone git@[example.org:12345]:some-organization/my-repository.git
error: failed to resolve address for [example.org:12345]: Name or service not known; class=Net (12)
$
```

Same when starting a new repository and adding a remote using `jj git remote add origin ...`:

```
$ jj git remote add origin git@[example.org:12345]:some-organization/my-repository.git
$ jj git remote list
origin git@[example.org:12345]:some-organization/my-repository.git
$ jj git fetch
error: failed to resolve address for [example.org:12345]: Name or service not known; class=Net (12)
```

I fail to find any useful hints on how to properly git the URL when using SSH and a non-standard port. Is this just something that no one has ever needed to do or is my search-foo just lacking? :-)

Kind Regards,
Johannes

**Comment by kastl-ars** [comment, marked as answer]
Aaaaand I can answer myself, it seems libgit2 only allows ports using the `ssh://` syntax.

This works:

```
jj git clone ssh://git@example.org:12345/some-organization/my-repository.git
```

**Reply by lmas** [reply]
Thank you for finding this!
